src/block/Beacon.php

```diff
@@ -73,18 +120,16 @@ public function onScheduledUpdate() : void{
 			$radius = (10 * $beaconLevel) + 10;
 			$effectDuration = 9 + (2 * $beaconLevel);
 
-			foreach($this->position->getWorld()->getPlayers() as $player){
-				if($player->getPosition()->distance($this->position) <= $radius){
-					if($primaryE === $secondaryE){
-						$effect = EffectIdMap::getInstance()->fromId($secondaryE);
-						if($effect instanceof Effect){
-							$player->getEffects()->add(new EffectInstance($effect, $effectDuration * 20, 1));
-						}
-						break;
+			if($primaryE === $secondaryE){
+				foreach($this->position->getWorld()->getPlayers() as $player){
```

Comment:
To improve the performance, as some people have said, I think you should also use getNearbyPlayers